Propuesta modelo relacional Prestamos.

alexJava2015 opened the pull request:
Se agrega imagen y archivo editable correspondiente a prestamos, los cuales pueden ser ejemplares o digitales. El préstamo se asigna a un usuario (Observar usuario_ID en tabla "PRESTAMO" se simula ya que en este modelo no existe como tal la tabla usuario).

Commit:
Propuesta modelo relacional Prestamos.
Se agrega imagen y archivo editable correspondiente a prestamos, los cuales pueden ser ejemplares o digitales. El préstamo se asigna a un usuario (Observar usuario_ID en tabla "PRESTAMO" se simula ya que en este modelo no existe como tal la tabla usuario).

jorgerdc reviewed (changes requested):
Muchas gracias por la propuesta. Anexo algunas observaciones.
* Falta modelar la entidad EJEMPLAR, su status  y su histórico.
* Observaciones  para la tabla PRESTAMO:
    * La fecha de entrega y la clave de acceso dependen de cada uno de los ejemplares que se
       solicitaron en el préstamo. Por lo tanto estos atributos no pueden ir en esta tabla.  Existe una relación M:N entre PRESTAMO y EJEMPLAR. En la tabla intermedia es donde deberían ir estos atributos.  La fecha fin también depende de cada ejemplar que se incluye en el préstamo.
    * No es necesaria la columna ENTREGA_EXPIRADA_BIT.  Este evento puede ser uno de los status del préstamo.
    * Falta la fecha de status en PRESTAMO.
    * El nombre del campo USUARIO_ID_FK debe ser USUARIO_RID para homologar. Esta convención es la que se usaría para hacer referencia a PKs de otros servicios.
    * Falta el campo para guardar el importe de la multa en caso de aplicar.
* Mejorar el nombre de la tabla STATUS. ¿Status de qué ?... Sería más adecuado STATUS_PRESTAMO.
* No me queda clara la idea de la tabla TIPO_LIBRO.  En realidad lo que hace falta es la referencia hacia la tabla LIBRO. Es decir,  para cada registro en EJEMPLAR  se necesita saber cual es el libro asociado de dicho ejemplar. Se requiere agregar el atributo LIBRO_RID.

Commit:
Observaciones de revisor.
Se agregan cambios en base a las observacaiones realizadas por el revisor, esta versión tiene nuevas entidades correspondientes a ejemplar, y modificación de en algunas de de las existentes.

jorgerdc reviewed (changes requested):
Anexo nuevas observaciones al modelo
Tabla EJEMPLAR.
- No me queda claro la idea del atributo path_ubicacion_ejemplar
- falta FK desde status_ejemplary su fecha de status
- falta el atributo biblioteca_rid que indica  la biblio en donde se encuentra el ejemplar..
-libro_rid debe ir en esta tabla  mas no en PRESTAMO.
-  Los atributos de las tablas de status deberían ser clave y descripción para homologar.
- Nombrar correctamente a los históricos: historico_status_prestamo , historico_status_ejemplar
- En el historico de ejemplr falta la fecha de status
- La tabla PRESTAMO_EJEMPLAR está mal diseñada.
  Esta tabla es intermedia ya que rompe una relación M:N  . Se definió una PK artificial y luego se anexa ejemplar_id como parte de la PK ?? esto es incongruente.  Faltan atributos, fecha_fin, fecha_devolucion.
Sospecho que hay problemas para representar una relación M:N...
